.github/workflows/pullrequest.yml

```diff
@@ -36,22 +35,10 @@ jobs:
           restore-keys: |
             ${{ runner.os }}-maven
 
-      - name: Maven operation with Sonar
```

Comment:
I wanted to first agree on a general pipeline and then, if the secret issue is solved for PRs, we might add a specific GH action for invoking Sonar... It doesn't need to be in the same Maven goal with the remaining build - you might call `sonar:sonar` explicit and pass the required arguments to it... Or create a `sonar` profile and just call this... At any rate I would like to build-up separate execution pipeline steps with separated GH actions... It is clearer to get 
[] build
[] i-test
[] sonar
executed and evaluated distinct instead of just getting "something failed".

Reply:
> It doesn't need to be in the same Maven goal with the remaining build

Completely agreed. As I believe you're aware off, it's been a relatively recent addition to our setup. 
Any enhancements to be made in the process are fine if you ask me.

Reply:
As you can see on the regular build I moved sonar to a separate step. For the PR build we can do the same...

Reply:
I think we should have a separated step for sonar as you described for PRs as well... the secret 'problem' is only for 'external' contributors while for any AxonIQ member, it would (and should) work